src/ServiceControl.AcceptanceTests/Recoverability/ExternalIntegration/When_a_failed_message_is_archived.cs

```diff
@@ -0,0 +1,150 @@
+﻿namespace ServiceControl.AcceptanceTests.Recoverability.ExternalIntegration
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using Infrastructure;
+    using NServiceBus;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.Settings;
+    using ServiceBus.Management.Infrastructure.Settings;
+    using NUnit.Framework;
+    using ServiceControl.Contracts;
+    using ServiceControl.MessageFailures;
+    using TestSupport;
+    using TestSupport.EndpointTemplates;
+    using Newtonsoft.Json;
+
+    class When_a_failed_message_is_archived : AcceptanceTest
+    {
+        [Test]
+        public async Task Should_publish_notification()
+        {
+            CustomConfiguration = config => config.OnEndpointSubscribed<MyContext>((s, ctx) =>
+            {
+                if (s.SubscriberReturnAddress.IndexOf("ExternalProcessor", StringComparison.OrdinalIgnoreCase) >= 0)
+                {
+                    ctx.ExternalProcessorSubscribed = true;
+                }
```

Comment:
```suggestion
                ctx.ExternalProcessorSubscribed = s.SubscriberReturnAddress.Contains("ExternalProcessor");
```

Reply:
Technically speaking this suggestion could reset the flag to false if a second subscriber exists.